pageserver/src/layered_repository/inmemory_layer.rs

```diff
@@ -265,9 +265,10 @@ impl Layer for InMemoryLayer {
     }
 }
 
-// Type alias to simplify InMemoryLayer::freeze signature
-//
-type SuccessorLayers = (Vec<Arc<dyn Layer>>, Option<Arc<InMemoryLayer>>);
+pub struct SuccessorLayers {
+    pub frozen_layers: Vec<(Arc<dyn Layer>, PathBuf)>,
```

Comment:
> `PathBuf`

I could not find a good way to extract bytes from `bookfile::Book` struct, but please point me to one: IMO, storing a few byte portions in our upload queue is better than reading the files.

Reply:
> Reading the files seems the right approach to me

Sure, I agree that there's no way around it in general, was trying to vet the bytes idea to see if it's possible to optimize it later with some fixed sized "hot" queue with a few hundred MBs of data.

> because you can always call layer.path() to get it

The issue with that is that this method has to return `-> Option<PathBuf>` due to in-memory layer existence, while the existing methods for two other layer kinds now return just `-> PathBuf` which requires quite notable code adjustments.

But now, thanks to your review, I've thrown away a bunch of junk code and can replace it with proper changes :)